src/app/@ansyn/ol/plugins/annotations/annotations.visualizer.ts

```diff
@@ -564,4 +528,125 @@ export class AnnotationsVisualizer extends EntitiesVisualizer {
 	public setIconSrc(src: string) {
 		this.iconSrc = src;
 	}
+
+	editFeature(featureId: any) {
+		let oldFeature = null;
+		let event = null;
+
+		if (this.edited) {
+			const { originalFeature } = this.edited;
+			this.removeInteraction('translateInteractionHandler');
+			oldFeature = originalFeature;
+		}
+
+		if (!oldFeature || featureId !== oldFeature.getId()) { // start editing
+			const originalFeature: olFeature = this.source.getFeatureById(featureId);
+			this.updateFeature(originalFeature.getId(), { editMode: true });
+			const labelFeature = this.createLabelFeature(originalFeature);
+
+			this.addInteraction('translateInteractionHandler', this.moveLabelInteraction(originalFeature, labelFeature));
+			event = {
+				originalFeature,
+				labelFeature
+			};
+			this.source.addFeature(labelFeature);
+		} else {
+			this.updateFeature(featureId, { editMode: false });
+			this.source.removeFeature(this.edited.labelFeature);
+		}
+		this.source.refresh();
+		this.events.onEditStart.next(event);
+	}
+
+	private createLabelFeature(feature: olFeature): olFeature {
+		const center = this.getCenterOfFeature(feature);
+		const entity = this.getEntity(feature);
+		const { mode: entityMode } = feature.getProperties();
+		const { label } = feature.getProperties();
+		const labelFeature = new olFeature({
+			geometry: label.geometry ? label.geometry : new olPoint(center.coordinates),
+		});
+		labelFeature.setStyle(new olStyle({
+			text: new olText({
+				font: entity.style.initial.label.font,
+				fill: new olFill({ color: entity.style.initial.label.fill }),
+				text: label.text,
+				offsetY: entityMode === 'Point' ? 30 : 0
+			})
+		}));
+
+		return labelFeature;
+	}
+
+	private moveLabelInteraction(originalFeature: olFeature, labelFeature: olFeature): olTranslate {
+		const translateInteraction = new olTranslate({
+			features: new olCollection([labelFeature]),
+			hitTolerance: 2
+		});
+		translateInteraction.on('translateend', (translateend) => {
+			const newCoord = this.geoJsonFormat.writeGeometryObject(translateend.features.item(0).getGeometry());
+			this.projectionService.projectAccurately(newCoord, this.iMap.mapObject).subscribe( (accuracyPoint) => {
```

Comment:
check if result is after onResetView or visualizer response. etc... ... 
async = sync problems

consider to use pipe( switchMap(...)).subscribe()